app/main.js

```diff
@@ -1,20 +1,26 @@
 /* eslint-env node */
-
 const electron = require("electron")
-const app = electron.app
-const BrowserWindow = electron.BrowserWindow
+const {app, BrowserWindow, ipcMain, Menu, shell} = electron
 const isDev = require("electron-is-dev")
-
+const { URL } = require("url")
+const defaultMenu = require("electron-default-menu")
 const updater = require("./updater")
 const logger = require("./logger")
 
+const {authorizeUser, fetchAccessToken} = require("./userAuthentication")
+
 let mainWindow
+let deepLinkURLOnReady = null
 
 logger.init()
 
 function createWindow () {
   logger.info("creating app window")
 
+  app.setAsDefaultProtocolClient("x-github-classroom")
+  const menu = defaultMenu(app, shell)
```

Comment:
Sets mac menu options so users can copy, paste and get default menu options in app.